packages/filebrowser-extension/src/index.ts

```diff
@@ -406,7 +424,17 @@ function addCommands(app: JupyterLab, tracker: InstanceTracker<FileBrowser>, bro
   });
 
   commands.addCommand(CommandIDs.showBrowser, {
-    execute: () => { app.shell.activateById(browser.id); }
+    execute: args => {
+      const path = args.path as string || '';
+      const browserForPath = getBrowserForPath(path);
+
+      if (Object.is(browser, browserForPath)) {
```

Comment:
This seems brittle to me. If I understand correctly, you have set it up this way so that if the browser widget is hosted within a wrapper node (as the GitHub browser does) it will still activate correctly. But if the alternative file browsers have any other kind of DOM hierarchy, this will fail.

I think we can accomplish this more generally by checking if the browser we find is *somewhere* in the hierarchy of items in the left/right area.

I am envisioning something like
```typescript
// Check for browser not found
if (!browserForPath) {
  return;
}
// Shortcut if we are using the main file browser
if (browser === browserForPath) {
  app.shell.activateById(browser.id);
  return;
} else {
  const areas = ['left', 'right'];
  for (let area of areas) {
    const it = app.shell.widgets(area);
    let widget = it.next();
    while (widget) {
      if (widget.contains(browserForPath)) {
        app.shell.activateById(widget.id);
        return;
      }
      widget = it.next();
    }
  }
}
```

Reply:
Your idea sounds nicely robust. I'll try implementing it. 

In truth, I was quite surprised when I was debugging `app.shell.activateById` and found that all of the problems I was having getting it to work with github boiled down to the fact that the filebrowser in the tracker has an `.id` of `github-filebrowser`, but the the filebrowser in `app.shell._leftHandler` has an `.id` of `github-file-browser`. I'm still unfamiliar with the majority of Jupyterlab's design/code, but I assume the different ids are intentional (and have something to do with the wrapper node you mentioned). Could I please impose on you to explain this aspect of Jupyterlab's design?

Reply:
Sorry for the confusion there. As I understand it (@jasongrout may be able to give some more context), the library we use for the low-level UI, phosphor, keeps track of which panels are mounted on it by an `id`. Phosphor does a bunch of logic about which things to show, how to move things around, etc. So in general, if you want to do things about hiding/showing panels, you need to refer to the `id` of the panel that is actually mounted in the layout, rather than one further down in the hierarchy.

The `github-file-browser` vs `github-filebrowser` is just an unfortunate similarity of names that are used for different things.